src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -690,6 +692,27 @@ BlockIO InterpreterCreateQuery::fillTableIfNeeded(const ASTCreateQuery & create)
             create.temporary ? context.getSessionContext() : context,
             context.getSettingsRef().insert_allow_materialized_columns).execute();
     }
+    /// If the query is CREATE LIVE VIEW [db.]table TO [db.]target_table,
+    /// then trigger ALTER LIVE VIEW [db.]table REFRESH to cause the target table prepopulation
+    if (!create.attach && create.is_live_view && !create.to_table_id.empty())
+    {
+        auto alter = std::make_shared<ASTAlterQuery>();
+        alter->is_live_view = true;
+        alter->database = create.database;
+        alter->table = create.table;
+        ASTPtr command_list = std::make_shared<ASTAlterCommandList>();
+        alter->set(alter->command_list, command_list);
+        auto command = std::make_shared<ASTAlterCommand>();
+        command->type = ASTAlterCommand::LIVE_VIEW_REFRESH;
+        alter->command_list->add(std::move(command));
+
+        if (create.temporary && !context.getSessionContext().hasQueryContext())
+            context.getSessionContext().makeQueryContext();
+
+        auto alter_interpreter = InterpreterAlterQuery(alter,
```

Comment:
It is a pretty unusual to alter table here.
Is it possible to move this logic inside `LiveView`? May be to `IStorage::sturtup()`?

Reply:
Hi Nicolai, if I remember correctly the reason why I could not do it in the `IStorage::startup()` is because the target table specified in the `TO` clause might not exist when `IStorage::startup` is called.